app/src/store/__tests__/registration.test.ts

```diff
@@ -0,0 +1,140 @@
+import { Middleware } from '@reduxjs/toolkit';
+import { createStore, Store } from '../app';
+import * as userStoreModule from '../user';
+import * as accessTokenStoreModule from '../voice/accessToken';
+import * as registrationStoreModule from '../voice/registration';
+import * as auth0 from '../../../__mocks__/react-native-auth0';
+import * as voiceSdk from '../../../__mocks__/@twilio/voice-react-native-sdk';
+
+let fetchMock: jest.Mock;
+
+jest.mock('../../../src/util/fetch', () => ({
+  fetch: (fetchMock = jest.fn().mockResolvedValue({
+    ok: true,
+    text: jest.fn().mockResolvedValue(undefined),
+  })),
+}));
+
+describe('registration', () => {
+  let store: Store;
+  const dispatchedActions: any[] = [];
+
+  beforeEach(() => {
+    dispatchedActions.splice(0);
+    const logAction: Middleware = () => (next) => (action) => {
```

Comment:
Ha, so this is the middleware we talked about? I don't have preference between this and jest mocks. I'll leave it up to you.

Reply:
Yep, this is it. Super simple so I'm not _too_ concerned about complicating actual app logic, and it helps me test tremendously.